akka-http-core/src/test/scala/akka/testkit/ExplicitlyTriggeredScheduler.scala

```diff
@@ -0,0 +1,75 @@
+package akka.testkit
+
+import java.util.concurrent.ThreadFactory
+
+import akka.actor.{ ActorSystem, Cancellable, Scheduler }
+import akka.event.LoggingAdapter
+import com.typesafe.config.Config
+
+import scala.annotation.tailrec
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration.{ Duration, FiniteDuration }
+
+/**
+ * For testing: scheduler that does not look at the clock, but must be progressed manually by calling `timePasses`.
+ *
+ * This is not entirely realistic: jobs will be executed on the test thread instead of using the `ExecutionContext`, but does
+ * allow for faster and less timing-sensitive specs..
+ */
+class ExplicitlyTriggeredScheduler(config: Config, log: LoggingAdapter, tf: ThreadFactory) extends Scheduler {
```

Comment:
Wouldn't all of the methods still need to be thread-safe?

Reply:
You're right - improved this.